FlightJournal.Web/Aprs/AircraftEventHandler.cs

```diff
@@ -24,14 +24,15 @@ public AircraftEventHandler(IAprsListener aprsListener, FlightContext db)
 
         private void OnAircraftTakeoff(object sender, AircraftEvent e)
         {
-            Plane p = _db.Planes.FirstOrDefault(x => x.Registration.ToLower() == e.Aircraft.Registration.ToLower());
+            Plane p = _db.Planes.FirstOrDefault(x => x.Registration.ToLower() == e.Aircraft.Registration.ToLower() 
+                                                     || x.CompetitionId.ToLower() == e.Aircraft.CompetitionId.ToLower()); // need comp. id due to non-stringent registrations...
             if (p == null)
             {
                 Log.Debug($"{nameof(AircraftEventHandler)}: starting {e.Aircraft} not in DB - ignored");
                 return;
             }
 
-            var flights = _db.Flights.Where(f => f.Plane.PlaneId == p.PlaneId && f.Departure == null && f.Landing == null);
+            var flights = _db.Flights.Where(f =>  !f.Deleted.HasValue && f.LastUpdated.Date == DateTime.Today && f.Plane.PlaneId == p.PlaneId && f.Departure == null && f.Landing == null);
```

Comment:
We need to double check that the f.Deleted.HasValue element is rendered directly to SQL and does not result in full result set back (should be logged in trace view)

Reply:
or simply check on null instead, I guess